### ISSUE TITLE
Parametrization-based test scheduling

### ISSUE THREAD

**Issue by Jongy** [pytest-dev/pytest-xdist#511]
I have a test matrix in which the same test suite is run across multiple environments. It's implemented using a fixture which is parametrized with the set of environments, and all tests use it.

I'd like to use xdist, but I couldn't find a test load scheduler that'd fit: All tests running on the same environment must execute sequentially (no specific order is required, of course, but they mustn't run concurrently).

After a quick look under `xdist/schedulers`, I patched `xdist.schedulers.loadfile.LoadFileScheduling._split_scope` to split based on the value of the parametrized fixture (which, luckily, appears in the `nodeid`). This works just fine, but now I'd like to do it without patching xdist in runtime :)

So first of all, if there's any existing way to schedule parametrized tests, I'd be happy to use it instead... And if not, I could use some design help on how my patchy scheduling can be written so it's incorporated into xdist (I'd be happy to create a PR, I think it's quite useful, it was also requested in the past, e.g #324).

I was thinking a `--dist=param/nodeid_regex/` argument, then it'd split based on the result.

**Comment by subha-py** [comment]
This looks like a probable solution, would love the moderator to comment on this.  @RonnyPfannschmidt

**Comment by Jongy** [comment]
I'm having second thoughts about it. The documentation of `--dist=each` says `each: send each test to all available environments` which sounds like it does what we need. However I've never used the the `--tx` or any other way of specifying specs, so I'm not sure if it's already covered or not...

**Comment by symonk** [comment]
you could probably implement your own scheduler and use it at runtime? without monkey patching the library, seems a more extensible way of managing things

**Comment by Jongy** [comment]
Yeah, the monkey patching was only for the PoC :) My idea is to add a new scheduler, I'm just not entirely sure this is the right approach.
